### ISSUE TITLE
Myst III: Exile movement/look issue

### ISSUE THREAD

**Issue by jasondude7116** [libretro/scummvm#16]
## Description
Myst III: Exile -  The game registers properly in the Scummvm core and will start/run. After the intro, when I get to the first screen, (a desert/canyon area with a bird flying in the background) I can't seem to rotate/move the view. Mouse and gamepad cursor is active. I can right click to release cursor from movement mode and move the cursor around the screen. Game seems to be running. Bird is flying, sound is playing, cursor is active. Just can not seem to move/rotate the view.

### Expected behavior
Mouse/cursor should rotate/move view when in movement mode.

### Actual behavior
Mouse/cursor does not move/rotate view in view/movement mode.

### Screenshots
[If applicable, add screenshots to help explain your problem]

### Steps to reproduce the bug
1. Load Myst III: Exile
2. After intro plays, user will be at starting screen with canyon view.
3. Try to move/rotate view, but can not.

### Version/Commit
[You can find this information under Information->Core Information / System Information]
- ScummVM core: 2.6.1 (Feb 8, 2023)
- RetroArch: v1.14.0/Git version: e3c92b0 (Dec 13, 2022)

### ScummVM game/engine
myst3-win

### Environment information
- Device: PC
- OS: Win 10

### Additional context
[Add any other context about the problem here]

**Comment by spleen1981** [comment]
should be fixed now.

**Comment by jasondude7116** [comment]
I downloaded the new core. I can move the view now (somewhat) , but there is still an issue.
When rotating the view, I can only move in a ..maybe 60 deg or so arc. This is same for up/down and left/right. If I get to the edge of the arc, then release the mouse from view mode (moving cursor on static screen) and move the mouse to the edge of the screen, then go to mouse look mode again, the arc changes it's center point....so that I'm viewing a different 60deg arc. So by releasing and unreleasing look mode and moving the cursor, I can eventually rotate 360deg on X axis.
I should be able to move mouse continually to left/right and be able to rotate 360deg. Instead, the look mode hits a "wall" at about 60deg either direction.

**Comment by i30817** [comment]
Yes, it's the same problem. Since we are on completely different OSes, using different drivers and i tested the problem still exists even if you change fullscreen to 'fake fullscreen'/aka: exclusive fullscreen, it's a bug on either the core or the retroarch treatment of mouse input beyond it's 'driver' setup.

I think the second is more likely, but i testing things on other cores that can use the mouse, like dosbox wouldn't go amiss. I think it might give a false assurance though, because dosbox games have a very limited resolution, most of them so...

**Comment by i30817** [comment]
I think it's retroarch. I tried it on a rotoscaped dos game i had installed here randomly, (versailes 1865), and besides requiring a bit of a setup to get into the situation and being slow as heck, i could reproduce the line behavior.

Here is a video. There i move up 'enough' to confuse retroarch, then try to move the cursor down again. See the unnatural line.


Note this is NOT versailles 1865 on scummvm, it's it on dosbox:

[Screencast from 12-02-2023 02:40:40.webm](https://user-images.githubusercontent.com/757900/218289940-7b670f30-9c61-4810-a859-4b7885b5bf85.webm)

**Comment by spleen1981** [comment]
New fix merged.

**Comment by jasondude7116** [comment]
Tested Myst 3. Same scenario with about 60deg arc of view. Can not rotate past that.

**Comment by spleen1981** [comment]
Did you test this commit e7fe82c4886ea4e7a964f9148cb7d76b90fe1edc ?

**Comment by jasondude7116** [comment]
I'm not familiar with some of this stuff. Under "System Information", there is "Git Version: e3c92b0". I'm assuming you are referring to that? I'm sorry if I'm being a bit confusing. What I did was download a new Scummvm core. (Which has Feb 12 build date) Are you saying that I need to get the latest version of Retroarch because an update was done to that? (I will be back at PC in about an hour or so.) I can do whatever I should have done then and test.

**Comment by spleen1981** [comment]
Retroarch version is ok, you need to check core version from Core Information (a string starting with "v2.6.1.4...." last seven digits are the short reference commit.

**Comment by jasondude7116** [comment]
Sorry for the delay. I finally figured out how to check the version number of the Scummvm core. I was not on the same commit. I downloaded the Scummvm core again, and it's now on the same version number as the commit. ROTATION IS WORKING!!! Played Myst 3 for a few minutes and all seems great so far. Sorry for the confusion, I'm learning. Thank you very much for your patience, time, and work! Have a good one.

TLDR: I'm stupid, I fixed some of my stupid and everything is working now as expected with the https://github.com/libretro/scummvm/commit/e7fe82c4886ea4e7a964f9148cb7d76b90fe1edc ? commit.

**Comment by i30817** [comment]
Is rotation working _just_ with the mouse for you? For me, retropad works, but only mouse doesn't.

If you're even using the mouse that is.

**Comment by jasondude7116** [comment]
Tested mouse, retropad, and keyboard. All working properly for rotating the view with https://github.com/libretro/scummvm/commit/e7fe82c4886ea4e7a964f9148cb7d76b90fe1edc

**Comment by i30817** [comment]
Were you able to rotate 360º in the initial view with the mouse? I wasn't.

_Just_ the mouse. Using the retropad/keyboard resets it.

**Comment by spleen1981** [comment]
It works properly even with mouse only on Win10 test, closing here.
